statistics.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 Airbus Defence and Space
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package godal
+
+// Statisitics on a given band.
+type Statistics struct {
+	Min, Max, Mean, Std float64
+}
+
+type statisticsOpts struct {
+	approx       int
+	bForce       int
+	errorHandler ErrorHandler
+}
+
+//StatisticsOption is an option that can be passed to Band.Statistics
+//
+//Available Statistics options are:
+// - Aproximate() to allow the satistics to be computed on overviews or a subset od all tiles.
+// - Force() default is 1. If Force() is passed, statistics will be returned if they have already been setted. If not they will computed.
+// - ErrLogger
+type StatisticsOption interface {
+	setStatisticsOpt(so *statisticsOpts)
+}
+
+type approxOkOption struct{}
+
+func (aoo approxOkOption) setStatisticsOpt(so *statisticsOpts) {
+	so.approx = 1
+}
+
+//StatisticsAproximate allows the statistics to be computed on overviews or a subset of all tiles.
+func StatisticsApproximate() interface {
```

Comment:
Reuse the Approximate() option from histogram.go (in the same way you are reusing ErrorHandler)